test/blackbox-tests/test-cases/directory-targets/cache-shared-subdir.t

```diff
@@ -0,0 +1,54 @@
+We create 2 directory targets which share a whole subdirectory.
+
+  $ export DUNE_CACHE_ROOT=$PWD/.cache
+  $ export DUNE_CACHE=enabled
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.11)
+  > (using directory-targets 0.1)
+  > EOF
+
+  $ cat > dune << EOF
+  > (rule
+  >  (target (dir d1))
+  >  (deps gen.sh)
+  >  (action
+  >   (no-infer
```

Comment:
It might be clearer to move `no-infer` around the `write-file`. As far as I can tell, you just don't want the d1/x target from being inferred.

Then, you can remove the `deps` field